ante/README.md

```diff
@@ -0,0 +1,159 @@
+
+For a  [global fee](https://github.com/cosmos/gaia/blob/82c4353ab1b04cf656a8c95d226c30c7845f157b/x/globalfee/types/params.go#L54-L99) to be valid:
+- fees have to be alphabetically sorted by denomination (denom)
+- fees have to have non-negative amount, with a valid and unique denom (i.e no duplicates). 
+
+Global fees allow denoms with zero coins or value.
+
+Zero value coins are used to define fee denoms, when the chain might does not charge fees. Each transaction (except bypass transactions) have to meet the following global fee requirements:
+- All of the paidfees' denoms (except zero coins) have to be a subset of the globalfee's denom set.
+- All paidfees' contain at least one denom that is present and greater than/or equal to the amount of the same denom in globalfee.
+
+### Query global fees
+You can query the network, to retreive the globalfee value, by running the following commands:
+```shell
+gaiad q globalfee minimum-gas-prices
+# or
+gaiad q params subspace globalfee MinimumGasPricesParam
+```
+### Empty global fees and default global fees
+When the global fee is not setup, the query will return an empty globalfee list: `minimum_gas_prices: []`. Gaiad will use `0uatom` as global fee in this case. This is due to the CosmosHub accepting uatom as fee denom by default.
+
+### Setting up global fees via Gov Proposals
+An example of setting up a global fee by a [gov proposals](https://hub.cosmos.network/main/governance/proposals/) is shown below.
+  
+```shell
+gov submit-legacy-proposal param-change proposal.json
+````
+A `proposal.json` example:
+```
+{
+  "title": "Global fees Param Change",
+  "description": "Update global fees",
+  "changes": [
+    {
+      "subspace": "globalfee",
+      "key": "MinimumGasPricesParam",
+      "value": [{"denom":"stake", "amount":"0.002"}, {"denom":"uatom", "amount": "0.001"}]
+    }
+  ],
+  "deposit": "1000stake"
+}
+```
+Please note: in the above "value" field, coins must sorted alphabetically by denom.
+
+### Parameter: min_gas_prices
+min_gas_prices are a node's further requirement of fees. Zero coins are removed from min_gas_prices when [parsing min_gas_prices](https://github.com/cosmos/cosmos-sdk/blob/3a097012b59413641ac92f18f226c5d6b674ae42/baseapp/options.go#L27), this is different from global fees.
+- If the `min_gas_prices` set a denom that is not global fees's denom set. This min_gas_prices denom will not be considered when paying fees.
+- If the `min_gas_prices` have a denom in global fees's denom set, and the  min_gas_prices are lower than global fees, the fee still need to meet the global fees.
+- If the `min_gas_prices` have a denom in global fees's denom set, and the  min_gas_prices are higher than global fees, the fee need to meet the min_gas_prices.
+
+### Fee Checks
+Global fees, min_gas_prices and the paid fees all allow zero coins setup. After parsing the fee coins, zero coins and paid fees will be removed from the min_gas_prices and paid fees. 
```

Comment:
I'm not sure what you mean here...is it just that parsing strips any zero coins?